src/jobs.rs

```diff
@@ -87,18 +121,29 @@ impl Job {
                 let slug = slug
                     .as_ref()
                     .ok_or_else(|| StatusCode::BAD_REQUEST.into_response())?;
-                Command::new(cmd)
-                    .arg("deploy")
-                    .arg(slug)
-                    .stdin(Stdio::null())
-                    .stdout(Stdio::piped())
-                    .stderr(Stdio::piped())
-                    .kill_on_drop(true)
-                    .spawn()
-                    .map_err(|e| {
-                        error!("failed to spawn process: {e}");
-                        StatusCode::INTERNAL_SERVER_ERROR.into_response()
-                    })?
+
+                match &toml {
+                    None => enarx_deploy(&cmd, slug)?,
+                    Some(toml) => {
+                        let slug = Slug::new(slug.clone())
+                            .ok_or_else(|| StatusCode::BAD_REQUEST.into_response())?;
+                        let wasm_bytes = slug
+                            .read("main.wasm")
+                            .await
+                            .map_err(|_| StatusCode::INTERNAL_SERVER_ERROR.into_response())?
+                            .bytes()
+                            .await
+                            .map_err(|_| StatusCode::INTERNAL_SERVER_ERROR.into_response())?;
```

Comment:
Are these errors really Internal Server Errors? Isn't the slug user input?

Internal Server Errors are only for when the Benefice server itself has an issue (e.g. a bug or it instance ran out of space and could not create a temporary file)